### PR TITLE
Use aligned padding in fixed buffer structures.

### PR DESCRIPTION
**Pull request by MoFtZ**
This PR also addresses the "issues" related to extremely large kernel programs described in #229.

### CODE REVIEW

File: Src/ILGPU/IR/Construction/Structures.cs [hunk 1]
```diff
@@ -86,40 +87,87 @@ partial class IRBuilder
...
             }
             return instanceBuilder.Seal();
         }
 
+        /// <summary>
+        /// Copies the remaining bytes to fill the structure fields.
+        /// </summary>
+        /// <param name="instanceBuilder">The current structure builder.</param>
+        /// <param name="location">The current location.</param>
+        /// <param name="rawFieldValue"></param>
```
Comment (m4rs-mt) [inline Src/ILGPU/IR/Construction/Structures.cs:113]: QUESTION: missing description?

File: Src/ILGPU/IR/Construction/Structures.cs [hunk 2]
```diff
@@ -86,40 +87,87 @@ partial class IRBuilder
...
+            object rawFieldValue,
+            int initialBytes,
+            int numBytes)
+        {
+            var handle = GCHandle.Alloc(rawFieldValue, GCHandleType.Pinned);
+            try
+            {
+                byte* ptr = (byte*)handle.AddrOfPinnedObject();
+                int i = initialBytes;
+                for (/* empty */; i < numBytes; /* empty */)
```
Comment (m4rs-mt) [inline Src/ILGPU/IR/Construction/Structures.cs:128]: Suggestion: `while` instead?

File: Src/ILGPU/IR/Construction/Structures.cs (hunk 1 above)
Comment (MoFtZ) [reply Src/ILGPU/IR/Construction/Structures.cs]: Fixed.

File: Src/ILGPU/IR/Construction/Structures.cs (hunk 2 above)
Comment (MoFtZ) [reply Src/ILGPU/IR/Construction/Structures.cs]: Fixed.

**Pull request merged**